frontend/appflowy_flutter/lib/workspace/presentation/home/menu/sidebar/workspace/_sidebar_workspace_item_list.dart

```diff
@@ -0,0 +1,209 @@
+                  _getUserInfo(),
+                  fontSize: 12.0,
+                  overflow: TextOverflow.ellipsis,
+                  color: Theme.of(context).hintColor,
+                ),
+                const Spacer(),
+                FlowyButton(
+                  useIntrinsicWidth: true,
+                  text: const FlowySvg(FlowySvgs.add_m),
+                  onTap: () {
+                    _showCreateWorkspaceDialog(context);
+                    PopoverContainer.of(context).closeAll();
+                  },
+                ),
+              ],
+            ),
+          ),
+          for (final workspace in workspaces) ...[
+            _WorkspaceMenuItem(
+              workspace: workspace,
+              userProfile: userProfile,
+              isSelected: workspace.workspaceId == currentWorkspace.workspaceId,
+            ),
+            const VSpace(4.0),
+          ],
+        ],
+      ),
+    );
+  }
+
+  String _getUserInfo() {
+    if (userProfile.email.isNotEmpty) {
+      return userProfile.email;
+    }
+
+    if (userProfile.name.isNotEmpty) {
+      return userProfile.name;
+    }
+
+    return LocaleKeys.defaultUsername.tr();
+  }
+
+  Future<void> _showCreateWorkspaceDialog(BuildContext context) async {
+    if (context.mounted) {
+      await NavigatorTextFieldDialog(
+        title: LocaleKeys.workspace_create.tr(),
+        value: '',
+        hintText: '',
+        autoSelectAllText: true,
+        onConfirm: (name, context) async {
+          final request = CreateWorkspacePB.create()..name = name;
+          final result = await UserEventCreateWorkspace(request).send();
+          final message = result.fold(
+            (s) => LocaleKeys.workspace_createSuccess.tr(),
+            (e) => '${LocaleKeys.workspace_createFailed.tr()}: ${e.msg}',
+          );
+          if (context.mounted) {
+            showSnackBarMessage(context, message);
+          }
+        },
+      ).show(context);
+    }
+  }
+}
+
+class _WorkspaceMenuItem extends StatelessWidget {
+  const _WorkspaceMenuItem({
+    required this.workspace,
+    required this.userProfile,
+    required this.isSelected,
+  });
+
+  final UserProfilePB userProfile;
+  final UserWorkspacePB workspace;
+  final bool isSelected;
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (_) => WorkspaceMemberBloc(
+        userProfile: userProfile,
+        workspace: workspace,
+      )
+        ..add(const WorkspaceMemberEvent.initial())
+        ..add(const WorkspaceMemberEvent.getWorkspaceMembers()),
+      child: BlocBuilder<WorkspaceMemberBloc, WorkspaceMemberState>(
+        builder: (context, state) {
+          final members = state.members;
+          // settings right icon inside the flowy button will
+          //  cause the popover dismiss intermediately when click the right icon.
+          // so using the stack to put the right icon on the flowy button.
+          return Stack(
```

Comment:
What about using `FlowyHover > Listener/GestureDetector > Row > [WorkspaceIcon, Column, _buildRightIcon()]` instead of a Stack?

Reply:
The problem is the FlowyHover. Tt rebuilds after being clicked, which causes the popover to dismiss immediately.